src/Client-SDK/Temporal.WorkflowClient/ClientApi.cs

```diff
@@ -0,0 +1,466 @@
+
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync<TWfArg, TSigArg>(string taskQueueMoniker, TWfArg workflowArgs, string signalName, TSigArg signalArgs) where TWfArg : IDataValue where TSigArg : IDataValue { return null; }
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync<TWfArg, TSigArg>(string taskQueueMoniker, TWfArg workflowArgs, string signalName, TSigArg signalArgs, CancellationToken cancelToken) where TWfArg : IDataValue where TSigArg : IDataValue { return null; }
+
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync(IWorkflowExecutionConfiguration workflowRunConfig, string signalName) { return null; }
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync(IWorkflowExecutionConfiguration workflowRunConfig, string signalName, CancellationToken cancelToken) { return null; }
+
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync<TSigArg>(IWorkflowExecutionConfiguration workflowRunConfig, string signalName, TSigArg signalArgs) where TSigArg : IDataValue { return null; }
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync<TSigArg>(IWorkflowExecutionConfiguration workflowRunConfig, string signalName, TSigArg signalArgs, CancellationToken cancelToken) where TSigArg : IDataValue { return null; }
+
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync<TWfArg>(IWorkflowExecutionConfiguration workflowRunConfig, TWfArg workflowArgs, string signalName) where TWfArg : IDataValue { return null; }
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync<TWfArg>(IWorkflowExecutionConfiguration workflowRunConfig, TWfArg workflowArgs, string signalName, CancellationToken cancelToken) where TWfArg : IDataValue { return null; }
+
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync<TWfArg, TSigArg>(IWorkflowExecutionConfiguration workflowRunConfig, TWfArg workflowArgs, string signalName, TSigArg signalArgs) where TWfArg : IDataValue where TSigArg : IDataValue { return null; }
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync<TWfArg, TSigArg>(IWorkflowExecutionConfiguration workflowRunConfig, TWfArg workflowArgs, string signalName, TSigArg signalArgs, CancellationToken cancelToken) where TWfArg : IDataValue where TSigArg : IDataValue { return null; }
+
+        public Task<TryGetResult<WorkflowRun>> TryStartNewRunWithSignalAsync(IWorkflowExecutionConfiguration workflowRunConfig, PayloadsCollection workflowArgs, string signalName, PayloadsCollection signalArgs, CancellationToken cancelToken) { return null; }
+        #endregion TryStartNewRunWithSignalAsync(..)
+
+
+        #region GetActiveOrStartNewRunAsync(..)
+        public Task<WorkflowRun> GetActiveOrStartNewRunAsync(string taskQueueMoniker) { return null; }
```

Comment:
I suspect it's just `WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE_FAILED_ONLY` but yeah, we don't need all of these shortcuts. We need the simplest thing that accepts all parameters for starting. We can add shortcuts later.

Reply:
if is not more or less racy than "get latest", as by the time it is used it may no longer be the latest.
in that way, any GetXxxRun is racy.
`GetActiveOrStartNew` says "if a run is running, get me that, otherwise start one".
`StartNew` says start a new run and give me an error if you find one is already running.

we can simplify for sure, but there are scenarios for both.